src/ol/dom.js

```diff
@@ -122,3 +122,32 @@ export function replaceChildren(node, children) {
     node.insertBefore(newChild, oldChild);
   }
 }
+
+let domImplementation_ = undefined;
+
+/**
+ * Register an external DOMImplementation. Can be used to supply a DOMImplementation
+ * in non-browser environments.
+ *
+ * @param {DOMImplementation} domImplementation A DOMImplementation.
+ * @api
+ */
+export function registerDOMImplementation(domImplementation) {
+  domImplementation_ = document;
```

Comment:
Do you mean to use the provided argument?

Reply:
Yes, that was a blunder and due to that my first iteration was registerDocument. Will fix.. but so would it be preferable to inject Document instead of DOMImplementation?

The reason for using the name `register` instead of `set` is because I thought this to be a bit similar to proj4 injection which is done with method `register` in the `proj4` module.